contracts/evmscript/EVMScriptRunner.sol

```diff
@@ -40,30 +39,44 @@ contract EVMScriptRunner is AppStorage, Initializable, EVMScriptRegistryConstant
 
         bytes4 sig = executor.execScript.selector;
         bytes memory data = abi.encodeWithSelector(sig, _script, _input, _blacklist);
-        require(address(executor).delegatecall(data), ERROR_EXECUTION_REVERTED);
 
-        bytes memory output = returnedDataDecoded();
-
-        emit ScriptResult(address(executor), _script, _input, output);
+        bytes memory output;
+        assembly {
+            let success := delegatecall(
+                gas,                // forward all gas
+                executor,           // address
+                add(data, 0x20),    // calldata start
+                mload(data),        // calldata length
+                0,                  // don't write output (we'll handle this ourselves)
+                0                   // don't write output
+            )
 
-        return output;
-    }
+            output := mload(0x40) // free mem ptr get
+            mstore(0x40, add(output, add(returndatasize, 0x20))) // free mem ptr set
```

Comment:
This was already in the code, but I think we are allocating 32 bytes more of memory than we need when the call is successful, as we only copy `returndatasize - 0x20` bytes.

Reply:
Actually it turns out we were allocating 64 bytes more:

- Assume the current free mem ptr is at 0x80
- `returndatasize` is 0x60
- We copy 0x40 (0x60 - 0x20) bits onto 0x80
- Next free pointer should be 0xc0 (0x80, 0xa0 are taken by the copy)